flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporter.java

```diff
@@ -159,17 +168,51 @@ void applyRemovals() {
 
     @Override
     public void notifyOfAddedMetric(Metric metric, String metricName, MetricGroup group) {
-        MetricGroup metricGroup = unwrap(group);
-        LOG.debug("Registered {} @ {}", metricName, metricGroup);
-        synchronized (this) {
-            metrics.computeIfAbsent(metricGroup, dummy -> new HashMap<>()).put(metricName, metric);
+        if (captureMode != CaptureMode.NONE) {
+            MetricGroup metricGroup = unwrap(group);
+            LOG.debug("Registered {} @ {}", metricName, metricGroup);
+            synchronized (this) {
+                metrics.computeIfAbsent(metricGroup, dummy -> new HashMap<>())
+                        .put(metricName, metric);
+            }
         }
     }
 
     @Override
     public void notifyOfRemovedMetric(Metric metric, String metricName, MetricGroup group) {
-        synchronized (this) {
-            removedGroups.add(unwrap(group));
+        if (captureMode != CaptureMode.NONE) {
+            synchronized (this) {
+                if (captureMode == CaptureMode.EAGER_RELEASE) {
+                    removeMetric(metricName, group);
+                } else if (captureMode == CaptureMode.LAZY_RELEASE) {
+                    removedGroups.add(unwrap(group));
```

Comment:
Nit: `unwrap(group)` is called on both if branches you can put it before the if-statements.